src/api/web/api.js

```diff
@@ -138,10 +147,74 @@ module.exports = function(config, auth, storage) {
     res.redirect(base);
   });
 
+  route.get('/sidebar/(@:scope/)?:package', function(req, res, next) {
+    let packageName = req.params.package;
+    if (req.params.scope) {
+      packageName = `@${req.params.scope}/${packageName}`;
+    }
+
+    storage.get_package({
+      name: packageName,
+      keepUpLinkData: true,
+      req,
+      callback: function(err, info) {
+        res.set('Content-Type', 'application/json');
+
+        if (!err) {
+          info.latest = info.versions[info['dist-tags'].latest];
+          let propertyToDelete = ['readme', 'versions'];
+
+          _.forEach(propertyToDelete, ((property) => {
+            delete info[property];
+          }));
+
+          let defaultGravatar = 'https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm';
+
+          if (typeof _.get(info, 'latest.author.email') === 'string') {
+            info.latest.author.avatar = generateGravatarUrl(info.latest.author.email);
+          } else {
+            // _.get can't guarantee author property exist
+            _.set(info, 'latest.author.avatar', defaultGravatar);
+          }
+
+          if (_.get(info, 'latest.contributors.length', 0) > 0) {
+            info.latest.contributors = _.map(info.latest.contributors, (contributor) => {
+                if (typeof contributor.email === 'string') {
+                  contributor.avatar = generateGravatarUrl(contributor.email);
+                } else {
+                  contributor.avatar = defaultGravatar;
+                }
+
+                return contributor;
+              }
+            );
+          }
+
+          res.end(JSON.stringify(info));
+        } else {
+          res.status(404);
+          res.end();
+        }
+      },
+    });
+  });
+
   // What are you looking for? logout? client side will remove token when user click logout,
   // or it will auto expire after 24 hours.
   // This token is different with the token send to npm client.
   // We will/may replace current token with JWT in next major release, and it will not expire at all(configurable).
 
   return route;
 };
+
+/**
+ * Generate gravatar url from email address
+ * @param {string} email
+ * @return {string} url
+ */
+function generateGravatarUrl(email) {
+  email = email.trim().toLocaleLowerCase();
+  let emailMD5 = crypto.createHash('md5').update(email).digest('hex');
```

Comment:
This one as well, `utils` is the right place

Reply:
Got it, refactoring tomorrow